src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -857,6 +847,42 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
             ProjectImportElement importAfter = traversalProject.CreateImportElement(@"$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\SolutionFile\ImportAfter\*");
             importAfter.Condition = @"'$(ImportByWildcardBeforeSolution)' != 'false' and exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\SolutionFile\ImportAfter')"; // Avoids wildcard perf problem
 
+            ProjectPropertyGroupElement directorySolutionPropsPropertyGroup = traversalProject.CreatePropertyGroupElement();
```

Comment:
Can you put a comment around this block?

Reply:
Added